src/openzaak/components/catalogi/admin/admin_views.py

```diff
@@ -112,14 +114,18 @@ def get_context_data(self, **kwargs):
             context["iotype_forms"] = iotype_forms
 
         besluittypen = self.request.session.get("besluittypen")
-        if besluittypen:
+        if besluittypen and "besluittype_forms" not in context:
+
             besluittypen_dict = {
                 obj.omschrijving: obj.pk
                 for obj in BesluitType.objects.filter(catalogus=catalogus)
                 .order_by("omschrijving", "-datum_begin_geldigheid")
                 .distinct("omschrijving")
             }
+            # sort alphabetically, save for use in post
             besluittypen = sorted(besluittypen, key=lambda x: x[0]["omschrijving"])
+            self.request.session["besluittypen"] = besluittypen
+
             besluittype_forms = BesluitTypeFormSet(
```

Comment:
Saves the new order so it is the same when the order fails. 

Is it possible to test the label of a form field with Django webtest or regular test? I cannot find info online.

Reply:
For regular Django forms you can check the label pretty straightforward:
```
my_form = response.context["form"]
my_field = my_form.fields["my_field"]
my_field.label
```
In this case `besluittype_forms` is a Formset, i.e. a list of forms, so if you know the index of your form, you can  check the labels of this particular form
For example you have 1 form in your formset, so:

```
besluittype_forms = response.context["besluittype_forms"]
besluittype_forms.forms[0].fields["exisiting"].label
```
you can also use `Formset.get_form_kwargs` method:
```
besluittype_forms = response.context["besluittype_forms"]
besluittype_forms.get_form_kwargs(0).labels
```

if you want to use webtest, then I think you need to search html for `label` tag with `response.html.find_all` or something similar